engine-rest/engine-rest/src/test/java/org/operaton/bpm/engine/rest/security/auth/impl/HttpBasicAuthenticationProviderTest.java

```diff
@@ -0,0 +1,109 @@
+package org.operaton.bpm.engine.rest.security.auth.impl;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.operaton.bpm.engine.rest.security.auth.impl.HttpBasicAuthenticationProvider.BASIC_AUTH_HEADER_PREFIX;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.ws.rs.core.HttpHeaders;
+import java.util.Base64;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.operaton.bpm.engine.IdentityService;
+import org.operaton.bpm.engine.ProcessEngine;
+import org.operaton.bpm.engine.rest.security.auth.AuthenticationResult;
+
+class HttpBasicAuthenticationProviderTest {
+
+  private static final String USER_ID = "user";
+
+  private static final String PASSWORD = "password";
+
+  private static final String CREDENTIALS =
+      Base64.getEncoder().encodeToString(String.format("%s:%s", USER_ID, PASSWORD).getBytes());
+
+  private static final String AUTHORIZATION_HEADER = BASIC_AUTH_HEADER_PREFIX + CREDENTIALS;
+
+  @Test
+  void testExtractAuthenticatedUserNoAuthHeader() {
+    HttpBasicAuthenticationProvider provider = new HttpBasicAuthenticationProvider();
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+    ProcessEngine engine = Mockito.mock(ProcessEngine.class);
```

Comment:
Maybe set up provider, request, engine as fields and initialize by a BeforeEach method? These three are always initialized the same in several methods.